### PR TITLE
Add Enterprise Hub Serverless API Notebook

### PR DESCRIPTION
**Pull request by andrewrreed**
# What does this PR do?

Adds the Serverless API notebook for Enterprise Hub section

## Who can review?

@MoritzLaurer @merveenoyan

### CODE REVIEW

**Comment by andrewrreed** [comment]
@merveenoyan @MoritzLaurer 

This PR is ready for review, however the only thing that would hold up merging is that there seems [to be an issue with suno/bark](https://huggingface.slack.com/archives/C016D661PAN/p1717451493501099) behind the Inference API. Might make sense to hold off until that is fixed before merging.

One other quick question - anyone know of a way to get the audio widget to be displayed / useable in section: "4. Generating Speech from Text"?

Thanks in advance for the review!!

**Review by merveenoyan — APPROVED** [review]
Very well put that I didn't find anything to add other than small nits!

**Comment by MoritzLaurer** [comment]
Great recipe @andrewrreed ! Particularly cool how the outputs from the different APIs are inputs for the next API! 
I've left a few additional comments.

File: notebooks/en/enterprise_cookbook_overview.md
Comment (MoritzLaurer) [inline notebooks/en/enterprise_cookbook_overview.md:14]: can you add a hyperlink to the section page here? (I tried to do this for my Argilla notebook, like this:
`## [Data Annotation with Argilla Spaces](enterprise_cookbook_argilla)`
But this actually doesn't seem to work)

File: notebooks/en/_toctree.yml
Comment (MoritzLaurer) [inline notebooks/en/_toctree.yml:72]: put your notebook above the argilla notebook to maintain the ordering in the overview file: https://huggingface.co/learn/cookbook/enterprise_cookbook_overview#data-annotation-with-argilla-spaces

**Review by stevhliu — COMMENTED** [review]
Cool examples of the Serverless Inference API! 👏

File: notebooks/en/enterprise_cookbook_overview.md
Comment (stevhliu) [reply notebooks/en/enterprise_cookbook_overview.md:14]: I think since it's a section header with its own anchor tag/link, you can't add another link like you would with regular text.

**Comment by andrewrreed** [comment]
<p>Great TIP :)</p>

---

 <a href='https://app.reviewnb.com/huggingface/cookbook/pull/105/discussion/#comment-2150837800'>View entire conversation</a> on ReviewNB<div id="RAICReplyForNotification-ReviewNBCommentContext-DoNotDelete"></div>

**Comment by andrewrreed** [comment]
<p>Good call - I added this language that's hopefully not to overwhelming for users:</p><p> </p><p>Next, you'll need to create a [User Access Token](<a href="https://huggingface.co/docs/hub/security-tokens" target="_blank">https://huggingface.co/docs/hub/security-tokens</a>). A token with <code>read</code> or <code>write</code> permissions will work, however, we highly encourage the use of fine-grained tokens. For this notebook, you'll need a fine-grained token with <code>Inference &gt; Make calls to the serverless Inference API</code> permissions.</p>

---

 <a href='https://app.reviewnb.com/huggingface/cookbook/pull/105/discussion/#comment-2150034839'>View entire conversation</a> on ReviewNB<div id="RAICReplyForNotification-ReviewNBCommentContext-DoNotDelete"></div>

**Comment by andrewrreed** [comment]
<p>It's my savior with VSCode notebooks :)</p>

---

 <a href='https://app.reviewnb.com/huggingface/cookbook/pull/105/discussion/#comment-2150034897'>View entire conversation</a> on ReviewNB<div id="RAICReplyForNotification-ReviewNBCommentContext-DoNotDelete"></div>

**Comment by andrewrreed** [comment]
<p>Just too subtle :)</p>

---

 <a href='https://app.reviewnb.com/huggingface/cookbook/pull/105/discussion/#comment-2150034942'>View entire conversation</a> on ReviewNB<div id="RAICReplyForNotification-ReviewNBCommentContext-DoNotDelete"></div>

**Comment by andrewrreed** [comment]
<p>Good call!</p>

---

 <a href='https://app.reviewnb.com/huggingface/cookbook/pull/105/discussion/#comment-2150262523'>View entire conversation</a> on ReviewNB<div id="RAICReplyForNotification-ReviewNBCommentContext-DoNotDelete"></div>

**Comment by andrewrreed** [comment]
<p>Interesting, it renders fine here... I guess it's just a limitation of our docs builder. I'm going to look into it a bit further.</p>

---

 <a href='https://app.reviewnb.com/huggingface/cookbook/pull/105/discussion/#comment-2150262755'>View entire conversation</a> on ReviewNB<div id="RAICReplyForNotification-ReviewNBCommentContext-DoNotDelete"></div>

**Comment by andrewrreed** [comment]
<p>The motivation for including this was to show the simplicity before introducing the client which abstracts things away. I think I'll leave it, but take Moritz's advice and convert to Python requests rather than curl</p>

---

 <a href='https://app.reviewnb.com/huggingface/cookbook/pull/105/discussion/#comment-2150837870'>View entire conversation</a> on ReviewNB<div id="RAICReplyForNotification-ReviewNBCommentContext-DoNotDelete"></div>

**Comment by andrewrreed** [comment]
<p>Yes, this is a good call. I swapped out <code>curl</code> for <code>requests</code></p>

---

 <a href='https://app.reviewnb.com/huggingface/cookbook/pull/105/discussion/#comment-2150262423'>View entire conversation</a> on ReviewNB<div id="RAICReplyForNotification-ReviewNBCommentContext-DoNotDelete"></div>

**Comment by andrewrreed** [comment]
<blockquote>I'd also probably slightly rephrase the sentence on 500 000 being available in the very beginning</blockquote><p>Replaced 500k with "thousands"</p><p> </p><blockquote>that explain how the serverless API works in the backend</blockquote><p>I do have some language just after the section that explains it. The goal was to introduce the simplicity and then explain how it worked. But I agree that adding some info about <code>list_deployed_models()</code> is a good idea. Will add that here!  </p>

---

 <a href='https://app.reviewnb.com/huggingface/cookbook/pull/105/discussion/#comment-2150262379'>View entire conversation</a> on ReviewNB<div id="RAICReplyForNotification-ReviewNBCommentContext-DoNotDelete"></div>

**Comment by MoritzLaurer** [comment]
While creating the dedicated endpoint notebook, I noticed that fine-grained token access to serverless API calls alone is not enough, but the fine-grained token also needs access to the model repositories for downloading the tokenizer/processor. In your case that's llama-3 and idefics2 I think. Maybe add a note on this to avoid people being confused when they cannot download the tokenizer. 

I might actually recommend using a read-only token for quick experimentation, otherwise users need to add each new model to their fine-grained token permissions where they need to download a tokenizer/processor. The fine-grained token should then be used in production. WDYT?

**Comment by andrewrreed** [comment]
Great catch @MoritzLaurer - I just pushed [a commit](https://github.com/huggingface/cookbook/pull/105/commits/97edf7c376c77c75f5fca5878d123a09a8753fe1) to add some language on this

**Comment by andrewrreed** [comment]
I think this is all ready to go now - feel free to merge @MoritzLaurer @merveenoyan @stevhliu!

**Review by stevhliu — APPROVED** [review]
Thanks for the addition :)

**Pull request merged**
